src/serializer/ResidualHeapVisitor.js

```diff
@@ -507,7 +507,9 @@ export class ResidualHeapVisitor {
       oldDelayedEntries = this.delayedVisitGeneratorEntries;
       this.delayedVisitGeneratorEntries = [];
       for (let { generator, entry } of oldDelayedEntries)
-        generator.visitEntry(entry, this.createGeneratorVisitCallbacks(generator));
+        this._withScope(generator, () => {
+          generator.visitEntry(entry, this.createGeneratorVisitCallbacks(generator));
```

Comment:
Bonus points if you can come up with a test case that shows the underlying issue without this change...